File: Domain/Activity.cs
using System;
using System.Collections.Generic;
// using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }  // default for DbContext to recognize as primary key
        // [Required]
        public string Title { get; set; }
        public DateTime Date  { get; set; }
        public string Description { get; set; }
        public string Venue { get; set; }
        public string City { get; set; }
        public string Category { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();
    }
}

File: client-app/src/models/activity.ts
import { Profile } from "./profile";

export interface Activity {
  'id': string;
  'title': string;
  'date': Date | null;
  'description': string;
  'venue': string;
  'city': string;
  'category': string;
  'hostUsername': string;
  'isCancelled': boolean;
  'isGoing': boolean;
  'isHost': boolean;
  'host'?: Profile;
  'attendees': Profile[];
}

export class Activity implements Activity {
  constructor(init?:ActivityFormValues) {
    Object.assign(this, init);
  }
}

export class ActivityFormValues {
  id?: string = undefined;
  title: string = '';
  date: Date | null = null;
  description: string = '';
  venue: string = '';
  city: string = '';
  category: string = '';

  constructor(activity?: ActivityFormValues) {
    if (activity) {
      this.id = activity.id;
      this.title = activity.title;
      this.date = activity.date;
      this.description = activity.description;
      this.venue = activity.venue;
      this.city = activity.city;
      this.category = activity.category;
    }
  }
}